src/obfuscation.ts

```diff
@@ -3,12 +3,8 @@ import * as SparkMD5 from 'spark-md5';
 
 import { PrecomputedFlag } from './interfaces';
 
-export function getMD5Hash(input: string): string {
-  return SparkMD5.hash(input);
-}
-
-function saltedHasher(salt: string) {
-  return (input: string) => getMD5Hash(salt + input);
+export function getMD5Hash(input: string, salt = ''): string {
+  return new SparkMD5().append(salt).append(input).end();
```

Comment:
❗ Going through string is buggy for some bytes.

Because UTF-8 employs multi-byte encoding for higher values, bytes like `"\xff"` are expanded to two bytes (`0xc3 0xbf`) and get hashed to `f3f7437e8c1f303fc6742d9368b36cf6` instead of the correct value of `00594fd4f42ba43fc1ca0427a0576295`.

We have to use either `appendBinary` (which corrects for this behavior) or array-based API.

```suggestion
  return new SparkMD5().appendBinary(salt).append(input).end();
```